tests/test_cli.py

```diff
@@ -115,57 +114,60 @@ def test_command_list_node(workdir, mocker):
     ('traintuple', ['--objective-key', 'foo', '--algo-key', 'foo', '--dataset-key', 'foo',
                     '--data-samples-path']),
     ('testtuple', ['--traintuple-key', 'foo', '--data-samples-path'])]
-)
+                         )
 def test_command_add(asset_name, params, workdir, mocker):
     method_name = f'add_{asset_name}'
 
-    with tempfile.NamedTemporaryFile(mode='w', suffix='.json') as file,\
-            mock_client_call(mocker, method_name, response={}) as m:
-        json.dump({}, file)
-        file.seek(0)
-
-        client_execute(workdir, ['add', asset_name] + params + [file.name])
+    with mock_client_call(mocker, method_name, response={}) as m:
+        json_file = workdir / "temp_file.json"
+        with open(str(json_file), 'w') as f:
+            json.dump({}, f)
```

Comment:
Minor, to take full advantage of the Pathlib library, could be simplified:
```python
json_file.write_text(json.dumps({}))
```

(it's already done like this line 128)

Same comment for next tests

Reply:
Yes, but JSON is different on this point, I already tried, unsuccessfully 🤔

Reply:
The main difference is to use `json.dumps(obj)`: it will return a JSON formatted string containing the serialized input `obj`.

You can run the following snippet in a python script:
```python
import pathlib
import json
f = pathlib.PosixPath("/tmp/foo")
f.write_text(json.dumps({}))
```
You can then check that the file `tmp/foo` exists and contains the expecting value.